### PR TITLE
one cookiecutter to rule them all

### PR DESCRIPTION
**Pull request by stloma**
Merging of the ZODB and SQLAlchemy cookiecutters into this one.

The user will be given options to select template engine and persistence backend.

### CODE REVIEW

**Comment by stevepiercy** [comment]
Please sign https://github.com/Pylons/pyramid-cookiecutter-starter/blob/latest/CONTRIBUTORS.txt

I'll review in the next couple days.

File: hooks/post_gen_project.py [hunk 1]
```diff
@@ -26,12 +28,62 @@ def delete_other_ext(directory, extension):
...
+
+    scaffold_directory = os.path.join(
+                WORKING, '{{cookiecutter.repo_name}}')
+
+    delete_other_files(scaffold_directory, prefix, rm_prefixes)
+
+
+def delete_other_files(directory, current_prefix, rm_prefixes):
+    """
+    Each backend has associated files in the cookiecutter, prefixed by it's
```
Comment (stevepiercy) [inline hooks/post_gen_project.py]: its

File: README.rst [hunk 2]
```diff
@@ -6,7 +6,7 @@ pyramid-cookiecutter-starter
         :target: https://travis-ci.org/Pylons/pyramid-cookiecutter-starter
         :alt: Master Travis CI Status
 
-A Cookiecutter (project template) for creating a Pyramid starter project using URL dispatch for routing and either Jinja2, Chameleon, or Mako for templating.
+A Cookiecutter (project template) for creating a Pyramid starter project using URL dispatch. Customizable options upon install are Jinja2, Chameleon, or Mako for templating and memory, SQLAlchemy, or ZODB for a persistent backend.
```
Comment (stevepiercy) [inline README.rst]: A Cookiecutter (project template) for creating a Pyramid starter project.
Customizable options upon install include choice of:
* template language (Jinja2, Chameleon, or Mako)
* persistent backend (none, SQLAlchemy with SQLite, or ZODB)
* mapping of URLs to routes (if selected persistent backend is "none" or "sqlalchemy" then URL dispatch, and if "zodb" then traversal)

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This looks great. Only a couple of minor changes, then I'll run through the matrix of options.

**Comment by stevepiercy** [comment]
Can we automate the build process across all the selected options of template language and persistent backend, for a total of 9 permutations? I don't know how to do that.

**Review by stevepiercy — APPROVED** [review]
Thank you! These changes are good. I still need to run through the 9 permutations.

**Comment by stloma** [comment]
No rush :) I'll begin looking through the build process as well.

File: tests/test_it.py [hunk 3]
```diff
@@ -33,6 +143,92 @@ def test_it(cookies, venv, capfd, template):
...
+    assert zodb_files == files
+
+    if WIN:
+        assert 'Scripts\\pserve' in out
+    else:
+        assert 'bin/pserve' in out
+
+    cwd = result.project.strpath
+
+    # this is a hook for executing scaffold tests against a specifick
```
Comment (stevepiercy) [inline tests/test_it.py]: specific

File: tests/test_it.py [hunk 4]
```diff
@@ -33,6 +143,92 @@ def test_it(cookies, venv, capfd, template):
...
+    assert sqlalchemy_files == files
+
+    if WIN:
+        assert 'Scripts\\pserve' in out
+    else:
+        assert 'bin/pserve' in out
+
+    cwd = result.project.strpath
+
+    # this is a hook for executing scaffold tests against a specifick
```
Comment (stevepiercy) [inline tests/test_it.py]: specific

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Spelling corrections

**Comment by stevepiercy** [comment]
@mmerickel @stloma has worked through all the fun issues on Windows. I'd appreciate your review to find any improvements.

**Review by stevepiercy — APPROVED** [review]

**Review by mmerickel — APPROVED** [review]
this is wild but I think you've done it in a very clean and maintainable way +1

**Comment by stevepiercy** [comment]
I'll deprecate the other CCs in their readmes on their master branches.

**Pull request merged**